openwisp_radius/api/views.py

```diff
@@ -1,8 +1,49 @@
+from django.contrib.auth.models import AnonymousUser
+from django.core.cache import cache
+from django_freeradius.api.views import AccountingView as BaseAccountingView
+from django_freeradius.api.views import AuthorizeView as BaseAuthorizeView
 from django_freeradius.api.views import BatchView as BaseBatchView
-from django_freeradius.api.views import accounting, authorize, postauth  # noqa
+from django_freeradius.api.views import PostAuthView as BasePostAuthView
+from django_freeradius.api.views import TokenAuthentication as BaseTokenAuthentication
+from rest_framework.exceptions import AuthenticationFailed
 
 from openwisp_users.models import Organization
 
+from ..models import OrganizationRadiusSettings
+
+
+class TokenAuthentication(BaseTokenAuthentication):
+    def authenticate(self, request):
+        id = None
+        token = None
+        if request.META.get('HTTP_AUTHORIZATION'):
+            headers = request.META.get('HTTP_AUTHORIZATION').split(',')
+            for header in headers:
+                try:
+                    id = header.split(' ')[1]
+                    token = header.split(' ')[2]
+                except IndexError:
+                    raise AuthenticationFailed('Token authentication failed')
+        elif request.GET.get('token') and request.GET.get('id'):
+            id = request.GET.get('id')
+            token = request.GET.get('token')
+
+        if id and token:
```

Comment:
alternatively here you could do:

```python
if not uuid or not token:
    raise AuthenticationFailed('Token authentication failed')

# etc.
```
This keeps the indentation level low so the code is easier to read (too many nested levels make it hard to read).